### PR TITLE
Add typings to most of the Redux logic

### PR DESCRIPTION
**Pull request by j-f1**
This will automatically flow types into `useSelector` and `dispatch` calls. Action (heh) item: port components to `useActions()` to get access to typed actions.

### CODE REVIEW

**Comment by j-f1** [comment]
Gonna reorg the commits here & open a second PR with just the modal types since they’re the most complex bit.

File: packages/loot-core/src/client/actions/budgets.ts [hunk 1]
```diff
@@ -1,12 +1,15 @@
 import { send } from '../../platform/client/fetch';
 import { getDownloadError, getSyncError } from '../../shared/errors';
+import type { Handlers } from '../../types/handlers';
 import * as constants from '../constants';
+import 'react-redux';
```
Comment (MatissJanis) [inline packages/loot-core/src/client/actions/budgets.ts:5]: oops?
```suggestion
```

File: packages/loot-core/src/mocks/budget.ts [hunk 2]
```diff
@@ -563,7 +562,7 @@ export async function createTestBudget(handlers) {
   await db.runQuery('DELETE FROM categories;');
   await db.runQuery('DELETE FROM category_groups');
 
-  let accounts: AccountEntity[] = [
+  let accounts: { name: string; offBudget?: 1; id?: string }[] = [
```
Comment (MatissJanis) [inline packages/loot-core/src/mocks/budget.ts:565]: **❓ question:** ‏do we even need a type here? Isn't it already implied from the data struct we provide it?

File: packages/desktop-client/src/components/Modals.tsx [hunk 3]
```diff
@@ -230,8 +218,6 @@ function Modals({
               key={name}
               modalProps={modalProps}
               month={options.month}
-              actions={actions}
-              isGoalTemplatesEnabled={isGoalTemplatesEnabled}
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/Modals.tsx:234]: **🔨 warning:** ‏oops?

File: packages/loot-core/src/mocks/budget.ts (hunk 2 above)
Comment (j-f1) [reply packages/loot-core/src/mocks/budget.ts:565]: Yes, because otherwise the `id` property won’t be inferred

**Review by MatissJanis — APPROVED** [review]
thanks!

**Pull request merged**